pkg/hubble/relay/pool/manager_test.go

```diff
@@ -704,6 +709,176 @@ func TestPeerManager(t *testing.T) {
+		wantStatus map[string]uint32
+	}{
+		{
+			name: "7 peers with all possible non-nil connectivity states",
+			pcBuilder: testutils.FakePeerClientBuilder{
+				OnClient: func() onClientFunc {
+					i := -1
+					return func(target string) (peerTypes.Client, error) {
+						return &testutils.FakePeerClient{
+							OnNotify: func(_ context.Context, _ *peerpb.NotifyRequest, _ ...grpc.CallOption) (peerpb.Peer_NotifyClient, error) {
+								cns := []*peerpb.ChangeNotification{
+									{
+										Name:    "peererino_uno",
+										Address: "192.0.1.1",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_dos",
+										Address: "192.0.1.2",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_tres",
+										Address: "192.0.1.3",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_cuatro",
+										Address: "192.0.1.4",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_cinco",
+										Address: "192.0.1.5",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_seis",
+										Address: "192.0.1.6",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+									{
+										Name:    "peererino_siete",
+										Address: "192.0.1.7",
+										Type:    peerpb.ChangeNotificationType_PEER_ADDED,
+									},
+								}
+								return &testutils.FakePeerNotifyClient{
+									OnRecv: func() (*peerpb.ChangeNotification, error) {
+										i++
+										switch {
+										case i >= len(cns):
+											once.Do(func() { close(done) })
+											return nil, io.EOF
+										default:
+											return cns[i], nil
+										}
+									},
+								}, nil
+							},
+							OnClose: func() error {
+								return nil
+							},
+						}, nil
+					}
+				}(),
+			},
+			ccBuilder: FakeClientConnBuilder{
+				OnClientConn: func() onClientConnFunc {
+					var i int
+					return func(target, hostname string) (poolTypes.ClientConn, error) {
+						return testutils.FakeClientConn{
+							OnGetState: func() connectivity.State {
+								states := []connectivity.State{
+									connectivity.Idle,
+									connectivity.Ready,
+									connectivity.Connecting,
+									connectivity.TransientFailure,
+									connectivity.Shutdown,
+								}
+								resultState := states[i%len(states)]
+								i = (i + 1) % 7
+								return resultState
+							},
+							OnClose: func() error {
+								return nil
+							},
+						}, nil
+					}
+				}(),
+			},
+			wantStatus: map[string]uint32{
+				"READY":             2,
+				"CONNECTING":        1,
+				"IDLE":              2,
+				"TRANSIENT_FAILURE": 1,
+				"SHUTDOWN":          1,
+				"NIL_CONNECTION":    0,
```

Comment:
nit: It might be a good idea to add a test case that actually returns a `NIL_CONNECTION`.  Currently we could just remove all of the `nilConnNum` counting logic and the tests will still pass.

Reply:
Sure, added a test case for that.